library/Zend/Mail/Storage/Part.php

```diff
@@ -102,10 +102,14 @@ public function __construct(array $params)
                 $this->headers = new Headers();
                 $this->headers->addHeaders($params['headers']);
             } else {
-                if (empty($params['noToplines'])) {
-                    Mime\Decode::splitMessage($params['headers'], $this->headers, $this->topLines);
+                if ($params['headers'] instanceof \Zend\Mail\Headers) {
+                    $this->headers = $params['headers'];
```

Comment:
Headers container should be cloned to avoid container sharing between several messages.